packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2982,6 +2982,14 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       }
     }
 
+    // Check for changes in viewId.
+    if (_hasInputConnection) {
+      final int newViewId = View.of(context).viewId;
```

Comment:
Does this also need to be done in initState? I'm worried that if you build an EditableText, and then without ever doing something to result in didChangeDependencies being called you change the viewId, then the EditableText will not be aware. Or maybe `textInputConfiguration` will be accessed and the dependency will be established there? Maybe it's worth a test if this isn't irrelevant in practice.

Reply:
As far as I can tell, we don't **_need_** to do this in `initState` but I'm open to any ideas that makes the code clearer and/or easier to understand.

Currently, the way it works is we "calculate" the `viewId` inside `TextInputConfiguration get textInputConfiguration` (which is accessed when opening a connection). From that point on, if `viewId` changes, we send an `updateConfig` message to let the engine know about that.

Reply:
Is `context` available in `initState`? Didn't we have an issue with that when we were investigating if platform views needed a viewId or not?

Reply:
> Is `context` available in `initState`? Didn't we have an issue with that when we were investigating if platform views needed a viewId or not?

This is a moot point now. We do not need `context` and `viewId` in `initState` at this time.